CONTRIBUTING.md

```diff
@@ -273,34 +280,12 @@ $ npm test
 
 and ensure all tests pass.
 
-### Squash Your Commits
-When you make a pull request, all of your changes need to be in one commit.
-
-If you have made more than one commit, then you will need to _squash_ your commits.
-
-To do this, see [Squashing Your Commits](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231).
-
-### Commit Message
```

Comment:
Lets keep this, we want to recommend the use and understanding of conventional commits. I agree that we have a npm script to do this, but an intro to this should be available somewhere IMHO.

Reply:
A couple hours after I made this PR, somebody got confused by this section: https://gitter.im/FreeCodeCamp/Contributors?at=5a443e6829ec6ac3118f11f8

The only thing that this section has that the PR list doesn't have is the template, but that part seems more confusing than helpful.

I'm not sure how to write this section in a way that it adds value and is not confusing like it was previously. Do you have any suggestions?

Reply:
I guess, its time we should have a dedicated `docs/` directory. And documentation related to the code-base in there. But that can be done later and progressively.

I am of the opinion on something along the lines of the guides repo's contributing guide. 

Its an excellent example.

For this lets do as you have suggested.